core/lib/src/response/named_file.rs

```diff
@@ -87,27 +104,55 @@ impl NamedFile {
 /// implied by its extension, use a [`File`] directly.
 impl<'r> Responder<'r, 'static> for NamedFile {
     fn respond_to(self, req: &'r Request<'_>) -> response::Result<'static> {
-        let mut response = self.1.respond_to(req)?;
-        if let Some(ext) = self.0.extension() {
+        if let Some(last_modified) = &self.modified {
+            if let Some(if_modified_since) = req.headers().get_one("If-Modified-Since") {
+                let if_modified_since = match parse_if_modified_since(if_modified_since) {
+                    Ok(if_modified_since) => if_modified_since,
+                    Err(_err) => return Response::build().status(Status::BadRequest).ok(),
```

Comment:
I think we should ignore the header in this case, instead of sending a 400:

> A recipient MUST ignore the If-Modified-Since header field if the
>   received field-value is not a valid HTTP-date, or if the request
>    method is neither GET nor HEAD.
(<https://tools.ietf.org/html/rfc7232#section-3.3>)

Reply:
Thanks for pointing out the section in the RFC 7232. I wasn't aware of it.